types/node/index.d.ts

```diff
@@ -2181,19 +2185,18 @@ declare module "url" {
         pathname?: string;
         port?: string | number;
         protocol?: string;
-        query?: string | null | { [key: string]: string | string[] };
+        query?: string | null | ParsedUrlQuery;
         search?: string;
         slashes?: boolean;
     }
 
-    export interface Url extends UrlObject {
+    export interface OutputUrlObject extends InputUrlObject {
         port?: string;
-        query?: any;
```

Comment:
This is already provided and typed correctly in `UrlObject` (now `InputUrlObject`)